pkg/tsdb/cloudwatch/services/log_groups.go

```diff
@@ -33,20 +34,29 @@ func (s *LogGroupsService) GetLogGroups(req resources.LogGroupsRequest) ([]resou
 			input.AccountIdentifiers = []*string{req.AccountId}
 		}
 	}
-	response, err := s.logGroupsAPI.DescribeLogGroups(input)
-	if err != nil || response == nil {
-		return nil, err
-	}
-
 	var result []resources.ResourceResponse[resources.LogGroup]
-	for _, logGroup := range response.LogGroups {
-		result = append(result, resources.ResourceResponse[resources.LogGroup]{
-			Value: resources.LogGroup{
-				Arn:  *logGroup.Arn,
-				Name: *logGroup.LogGroupName,
-			},
-			AccountId: utils.Pointer(getAccountId(*logGroup.Arn)),
-		})
+
+	for {
+		input.NextToken = nextToken
+		response, err := s.logGroupsAPI.DescribeLogGroups(input)
+		if err != nil || response == nil {
+			return nil, err
+		}
+
+		for _, logGroup := range response.LogGroups {
+			result = append(result, resources.ResourceResponse[resources.LogGroup]{
+				Value: resources.LogGroup{
+					Arn:  *logGroup.Arn,
+					Name: *logGroup.LogGroupName,
+				},
+				AccountId: utils.Pointer(getAccountId(*logGroup.Arn)),
+			})
+		}
+
+		if !req.ListAllLogGroups || response.NextToken == nil {
```

Comment:
Why do we have both req.ListAllLogGroups and a req.Limit?

Reply:
Limit is the maximum items per page, and if ListAllLogGroups is true we'll iterate over all pages. If false, only the first page.